com.sap.cloud.lm.sl.mta/src/main/java/com/sap/cloud/lm/sl/mta/parsers/v2/ModuleParser.java

```diff
@@ -46,20 +62,44 @@ public Module parse() throws ParsingException {
         return builder.build();
     }
 
+    protected String getName() {
+        return getStringElement(NAME);
+    }
+
+    protected String getType() {
+        return getStringElement(TYPE);
+    }
+
+    protected String getDescription() {
+        return getStringElement(DESCRIPTION);
+    }
+
+    protected Map<String, Object> getProperties() {
+        return getMapElement(PROPERTIES);
+    }
+
     protected String getPath() {
         return getStringElement(PATH);
     }
 
     protected Map<String, Object> getParameters() {
         return getMapElement(PARAMETERS);
     }
-
     protected List<ProvidedDependency> getProvidedDependencies2() {
-        List<ProvidedDependency> providedDependencies = ListUtil.cast(getProvidedDependencies1());
+        List<ProvidedDependency> providedDependencies = ListUtil.cast(getProvidedDependencies());
         return getAllProvidedDependencies(providedDependencies);
     }
+    
+    protected List<ProvidedDependency> getProvidedDependencies() {
+        return getListElement(PROVIDES, new ListParser<ProvidedDependency>() {
+            @Override
+            protected ProvidedDependency parseItem(Map<String, Object> map) {
+                return getProvidedDependencyParser(map).setUsedValues(usedProvidedDependencyNames)
+                    .parse();
+            }
+        });
```

Comment:
I guess that this was the original v1. coding, but it could be made more readable